src/main/java/Duke.java

```diff
@@ -1,10 +1,214 @@
-public class Duke {
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+import java.io.IOException;
+
+/**
+ * Driver class of Duke chatbot, also saves on startup and loads on exit.
+ */
+public class Duke extends Application {
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    private Image user;
+    private Image duke;
+
+    /**
+     * Storage for loading task list, and to save right before exiting Duke.
+     */
+    private Storage storage;
+
+    /**
+     * TaskList to keep track of existing tasks.
+     */
+    private TaskList tasks;
+
+    /**
+     * Ui to provide user interface printing.
+     */
+    private Ui ui;
+
+    /**
+     * Empty constructor of Duke.
+     */
+    public Duke() {
+        this.user = new Image(this.getClass().getResourceAsStream("images/DaUser.png"));
+        this.duke = new Image(this.getClass().getResourceAsStream("images/DaDuke.png"));
+    }
+
+    /**
+     * Constructor of Duke.
+     *
+     * @param filePath String representing file path of save file of tasks.
+     */
+    /*
+    public Duke(String filePath) {
```

Comment:
i understand having commented code like that when trying out different stuff, just remember to clean it up when you're done!